app/src/main/java/com/android/wardrobeManager/ui/images/filters/Checker.java

```diff
@@ -0,0 +1,39 @@
+package com.android.wardrobeManager.ui.images.filters;
+
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.PorterDuff;
+import android.graphics.PorterDuffXfermode;
+
+import com.android.wardrobeManager.ui.images.DesignFilterManager;
+
+public class Checker implements DesignFilterManager.DesignFilter {
+    int checkerSize = 40;
+    int colorIndex = 0;
+
+    public Bitmap filter(Bitmap bitmap, Bitmap ref, int[] colors) {
+        bitmap = ref.copy(ref.getConfig(), true);
+        Canvas canvas = new Canvas(bitmap);
+        Paint paint = new Paint(Paint.ANTI_ALIAS_FLAG);
+        paint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN));
+        paint.setColor(colors[0]);
+        canvas.drawRect(0, 0, bitmap.getWidth(), bitmap.getHeight(), paint);
+
+        for (int i = 0; i < bitmap.getHeight(); i += checkerSize) {
+            for (int j = 0; j < bitmap.getWidth(); j += checkerSize) {
+                if ((i / checkerSize) % 2 == (j / checkerSize) % 2) {
+                    if (colors.length == 1) {
+                        paint.setColor(Color.WHITE);
+                    } else {
+                        colorIndex = colorIndex % (colors.length - 1) + 1;
+                        paint.setColor(colors[colorIndex]);
```

Comment:
This is a little confusing... The following is a little more obvious
`colorIndex = (colorIndex + 1) % (colors.length - 1);
                        paint.setColor(colors[colorIndex + 1]);`
Maybe add a small comment before it explaining that the colors drawn in the rectangles are all the colors except the first one.